src/eventProcessor/GoogleAnalyticsEventProcessor.js

```diff
@@ -50,12 +64,57 @@ class GoogleAnalyticsEventProcessor {
   constructor({
       'api_secret': apiSecret,
       'measurement_id': measurementId,
-      'measurement_url': measurementUrl = DEFAULT_MEASUREMENT_URL,
-      'client_id_expires': clientIdExpires = 2 * 365 * 24 * 60 * 60,
-      'automatic_params': userAutomaticParams = [],
+      'measurement_url': measurementUrl,
+      'client_id_expires': clientIdExpires,
+      'automatic_params': userAutomaticParams,
     } = {}) {
-    // TODO(kjgalvan):: log error if measurementUrl is default value and
-    // either apiSecret or measurementId are undefined.
+    if (apiSecret !== undefined && typeof apiSecret !== 'string') {
+      logging.log(
+        'Provided API Secret is not a valid string.',
+        logging.LogLevel.ERROR
+      );
+      apiSecret = undefined;
+    }
+    if (measurementId !== undefined && typeof measurementId !== 'string') {
+      logging.log(
+        'Provided Measurement ID is not a valid string.',
+        logging.LogLevel.ERROR
+      );
+      measurementId = undefined;
+    }
+    if (typeof measurementUrl !== 'string') {
+      if (measurementUrl !== undefined) {
+        logging.log(
+          'Provided Measurement URL is not a valid string. ' +
+              'Using default URL instead.',
+          logging.LogLevel.ERROR
+        );
+      }
+      measurementUrl = getDefaultMeasurementUrl();
+    }
+    if (clientIdExpires === undefined || isNaN(Number(clientIdExpires))) {
+      if (clientIdExpires !== undefined) {
+        logging.log(
+          'Provided Client ID Expires is not a valid number. ' +
+              'Using default of two years instead.',
+          logging.LogLevel.ERROR
+        );
+      }
+      clientIdExpires = 2 * 365 * 24 * 60 * 60;
+    } else {
+      clientIdExpires = Number(clientIdExpires);
+    }
+    if (Object.prototype.toString.call(userAutomaticParams) !==
```

Comment:
While this is correct, could you please check what browser versions we must support. There's a chance that Array.isArray will do